src/System.Net.Sockets/tests/FunctionalTests/DisposedSocketTests.netcoreapp.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public partial class DisposedSocket
+    {
+        //[ActiveIssue(37044, TestPlatforms.AnyUnix)]
+        [Fact]
+        public void NonDisposedSocket_SafeHandlesCollected()
+        {
+            List<WeakReference> handles = CreateHandles();
+            RetryHelper.Execute(() =>
+            {
+                GC.Collect();
+                Assert.Equal(0, handles.Count(h => h.IsAlive));
+            });
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static List<WeakReference> CreateHandles()
+        {
+            var handles = new List<WeakReference>();
```

Comment:
@stephentoub don't we need something 'async' to get into the static SocketAsyncEngine Dictionary?

Reply:
Yup, forgot that part.  Thanks.